src/com/facebook/buck/android/GenerateManifestStep.java

```diff
@@ -86,35 +83,47 @@ public StepExecutionResult execute(ExecutionContext context) {
 
     File skeletonManifestFile =
         filesystem.getPathForRelativeExistingPath(skeletonManifestPath).toAbsolutePath().toFile();
+    BuckEventAndroidLogger logger = new ManifestMergerLogger(context.getBuckEventBus());
 
-    ICallback callback = codename -> BASE_SDK_LEVEL;
-
-    IMergerLog log = MergerLog.wrapSdkLog(new BuckEventAndroidLogger(context.getBuckEventBus()));
-
-    ManifestMerger merger = new ManifestMerger(log, callback);
-
-    File outManifestFile = outManifestPath.toFile();
-    if (!merger.process(
-        outManifestFile,
-        skeletonManifestFile,
-        Iterables.toArray(libraryManifestFiles, File.class))) {
-      throw new HumanReadableException("Error generating manifest file");
-    }
+    MergingReport mergingReport =
+        mergeManifests(skeletonManifestFile, libraryManifestFiles, logger);
 
+    String xmlText = mergingReport.getMergedDocument(MergingReport.MergedManifestKind.MERGED);
     if (context.getPlatform() == Platform.WINDOWS) {
       // Convert line endings to Lf on Windows.
-      try {
-        String xmlText = Files.toString(outManifestFile, Charsets.UTF_8);
-        xmlText = xmlText.replace("\r\n", "\n");
-        Files.write(xmlText.getBytes(Charsets.UTF_8), outManifestFile);
-      } catch (IOException e) {
-        throw new HumanReadableException("Error converting line endings of manifest file");
-      }
+      xmlText = xmlText.replace("\r\n", "\n");
+    }
+    try {
+      filesystem.writeContentsToPath(xmlText, outManifestPath);
+    } catch (IOException e) {
+      throw new HumanReadableException(e, "Error writing manifest file");
     }
 
     return StepExecutionResult.SUCCESS;
   }
 
+  private MergingReport mergeManifests(
+      File mainManifestFile, List<File> libraryManifestFiles,
+      BuckEventAndroidLogger logger) {
+    try {
+      MergingReport mergingReport = ManifestMerger2
+          .newMerger(mainManifestFile, logger, ManifestMerger2.MergeType.APPLICATION)
+          .withFeatures(
+              ManifestMerger2.Invoker.Feature.NO_PLACEHOLDER_REPLACEMENT,
+              ManifestMerger2.Invoker.Feature.REMOVE_TOOLS_DECLARATIONS,
+              ManifestMerger2.Invoker.Feature.SKIP_BLAME)
+          .addLibraryManifests(Iterables.toArray(libraryManifestFiles, File.class))
+          .merge();
+      if (mergingReport.getResult().isError()) {
+        mergingReport.log(logger);
+        throw new HumanReadableException("Error generating manifest file");
```

Comment:
This is still showing up with no indication of what the actual error is.  I think we need to display all available diagnostic info in the console to make it easier for engineers (and me) to debug their build failures.